### PR TITLE
feat(gradle-inspector): Allow to customize the Java home for analysis

### PR DESCRIPTION
**Pull request by sschuberth**
Note that `setJavaHome()` [1] set the Java home to "to use for the Gradle process". However, as by default (unless toolchains are used [2]) "Gradle will compile Java code to the language level of the JVM running Gradle" [3], this will impact the Java target of the build.

[1]: https://docs.gradle.org/current/javadoc/org/gradle/tooling/ConfigurableLauncher.html#setJavaHome(java.io.File)
[2]: https://docs.gradle.org/8.10/userguide/building_java_projects.html#sec:compiling_with_toolchain
[3]: https://docs.gradle.org/8.10/userguide/building_java_projects.html#sec:java_cross_compilation

### CODE REVIEW

**Comment by sschuberth** [comment]
@mnonnenmacher could you give this a try with  one of your projects that require to be analyzed with a different Java version than ORT itself was built with?

**Comment by mnonnenmacher** [comment]
> @mnonnenmacher could you give this a try with one of your projects that require to be analyzed with a different Java version than ORT itself was built with?

I could not quickly test with on of our internal projects, so I wanted to do some tests with our test projects in `plugins/package-managers/gradle/src/funTest/assets/projects/synthetic`. However, when I add `.ort.yml` files like this:

```yaml
analyzer:
  enabled_package_managers: ["GradleInspector"]
```

I always get this error (with ORT from the main branch, without your change):

```
Exception in thread "main" java.lang.IllegalArgumentException: No DependencyGraph for package manager 'Gradle' available.
        at org.ossreviewtoolkit.model.DependencyGraphNavigator.graphForManager(DependencyGraphNavigator.kt:97)
        at org.ossreviewtoolkit.model.DependencyGraphNavigator.directDependencies(DependencyGraphNavigator.kt:48)
        at org.ossreviewtoolkit.analyzer.AnalyzerResultBuilderKt.resolvePackageManagerDependencies(AnalyzerResultBuilder.kt:117)
        at org.ossreviewtoolkit.analyzer.AnalyzerResultBuilderKt.access$resolvePackageManagerDependencies(AnalyzerResultBuilder.kt:1)
        at org.ossreviewtoolkit.analyzer.AnalyzerResultBuilder.build(AnalyzerResultBuilder.kt:51)
        at org.ossreviewtoolkit.analyzer.AnalyzerState.buildResult(Analyzer.kt:257)
```

I wanted to test what happens when I downgrade the test project to Gradle 7.0.1 which is not compatible with Java 17 (https://docs.gradle.org/current/userguide/compatibility.html) and then add the following to the `.ort.yml` to check if it makes a difference:

```yaml
analyzer:
  enabled_package_managers: ["GradleInspector"]
  package_managers:
    Gradle:
      options:
        javaHome: "/usr/lib/jvm/java-11-openjdk-amd64"
```

However, I cannot even get the project analyzed with the `GradleInspector` without any modifications, so I'm a little lost right now.

Btw, when I downgrade Gradle to 7.0.1 and use the old `Gradle` analyzer, I get errors like this:

```
        Caused by: GroovyBugError: BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 61
            Caused by: IllegalArgumentException: Unsupported class file major version 61
```

**Comment by mnonnenmacher** [comment]
Regarding the `No DependencyGraph for package manager 'Gradle' available.` error, could it be that something broke with the migration of `GradleInspector` to the dependency graph?

The `DependencyGraphNavigator` is looking for a graph like so:

```kotlin
requireNotNull(graphs[manager]) { "No DependencyGraph for package manager '$manager' available." }
```

But maybe the graph was stored with the key "GradleInspector" instead of "Gradle" and this is not discovered by the tests which work on individual `ProjectAnalyzerResult`s instead of `OrtResult`s?

**Comment by mnonnenmacher** [comment]
With #9085 merged I did another test with the `gradle-library` test project. I have downgraded the project to Gradle 6.9.4 and added the following `.ort.yml`:

```yaml
analyzer:
  enabled_package_managers: [ "GradleInspector" ]
```

When I analyze this, I get errors like this for each project:

```
13:57:53.270 [DefaultDispatcher-worker-1] ERROR org.ossreviewtoolkit.analyzer.PackageManager - GradleInspector failed to resolve dependencies for path 'lib/build.gradle': BuildException: Could not fetch model of type 'OrtDependencyTreeModel' using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-6.9.4-bin.zip'.
Caused by: LocationAwareException: Initialization script '/home/xxx/.ort/tools/GradleInspector/init.gradle'
Could not compile initialization script '/home/xxx/.ort/tools/GradleInspector/init.gradle'.
    Caused by: ContextualPlaceholderException: Could not compile initialization script '/home/xxx/.ort/tools/GradleInspector/init.gradle'.
        Caused by: PlaceholderException: startup failed:
        General error during semantic analysis: Unsupported class file major version 61
    
        java.lang.IllegalArgumentException: Unsupported class file major version 61
                at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:196)
```

Then I changed the `.ort.yml` to:

```yaml
analyzer:
  enabled_package_managers: [ "GradleInspector" ]
  package_managers:
    GradleInspector:
      options:
        javaHome: "/usr/lib/jvm/java-17-openjdk-amd64"
```

Now the analysis works!

If I change back to Java 17 it still works until I delete the `~/.gradle/caches/6.9.4/` directory, I guess because it caches the compiled init script.

File: model/src/test/kotlin/config/AnalyzerConfigurationTest.kt [hunk 1]
```diff
@@ -138,7 +138,7 @@ class AnalyzerConfigurationTest : WordSpec({
                 packageManagers = mapOf(
                     "Gradle" to PackageManagerConfiguration(
                         mustRunAfter = listOf("Npm"),
-                        options = mapOf("gradleVersion" to "7.6.1")
+                        options = mapOf("gradleVersion" to "7.6.1", "javaHome" to "/path/to/java/home")
```
Comment (mnonnenmacher) [inline model/src/test/kotlin/config/AnalyzerConfigurationTest.kt:141]: This should be an option of "GradleInspector", not "Gradle" (same below). Adding it here as a "Gradle" option could confuse someone.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

**Comment by mnonnenmacher** [comment]
It would be good to add a test case for this with a Gradle 6.9.4 test project, but this would require that we add a Java 11 JDK to the Docker image first to be able to run it on CI.

File: model/src/test/kotlin/config/AnalyzerConfigurationTest.kt (hunk 1 above)
Comment (sschuberth) [reply model/src/test/kotlin/config/AnalyzerConfigurationTest.kt:141]: Agreed. Actually, I was just adding it here "for fun" to further show that replaced options do not discard other existing options. But for simplicity, I'm inclined to just drop this change. Any objections?

Comment (sschuberth) [reply model/src/test/kotlin/config/AnalyzerConfigurationTest.kt:141]: Scratch that, I just made this particular test use "GradleInspector", which is what you suggested I believe, I just misread it initially.

**Review by mnonnenmacher — APPROVED** [review]

**Comment by sschuberth** [comment]
Merging despite the known `NuGetFunTest` flakyness.

**Pull request merged**